### PR TITLE
add run_docker.sh script

### PR DESCRIPTION
**Pull request by facaiy**
Based on #325 . Related with #294.

@seanpmorgan @av8ramit Hi, does it look complicated?

```bash
bash tools/run_docker.sh -c 'make code-format' 
# Github checkers
bash tools/run_docker.sh -c 'make sanity-check'  # Ubuntu Sanity Check
bash tools/run_docker.sh -s linux -d cpu -p py2 -c 'make unit-test'  # Ubuntu Python2
bash tools/run_docker.sh -s linux -d cpu -p py3 -c 'make unit-test'  # Ubuntu Python3
bash tools/run_docker.sh -s linux -d gpu -p py2 -c 'make gpu-unit-test'  # Ubuntu GPU Python2
bash tools/run_docker.sh -s linux -d gpu -p py3 -c 'make gpu-unit-test'  # Ubuntu GPU Python3
```

### CODE REVIEW

File: tools/run_docker.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        ;;
+esac
+
+
+DOCKER_CMD="ln -sf ${PYTHON_LIB} /usr/bin/python && ${COMMAND}"
+echo "Docker image: ${DOCKER_IMAGE}"
+echo "Docker command: ${DOCKER_CMD}"
+docker run ${DOCKER_OPTS}                   \
+    --network=host                          \
+    --rm -v ${ROOT_DIR}:/addons -w /addons  \
```
Comment (facaiy) [inline tools/run_docker.sh]: We mount the current dir to docker container, so is it safe to run two and more checkers at the same time?

**Comment by seanpmorgan** [comment]
LGTM... just want to confirm `nvidia-docker` is installed on the GCP image that this runs on. It'll also require new commands to be ran so until that is done the current CI test on this is not valid.

File: tools/run_docker.sh (hunk 1 above)
Comment (seanpmorgan) [reply tools/run_docker.sh]: My understanding was that these were ran in 4 separate GCP instances... but I could be incorrect.

**Comment by facaiy** [comment]
@av8ramit Ami, can you take a look? Hopefully it would work in GCP instances :-)

**Comment by av8ramit** [comment]
Sorry if I'm missing something, I glanced over this. What is invoking `run_docker.sh`? The internal jobs that run invoke `addons_gpu.sh`?

**Comment by seanpmorgan** [comment]
> Sorry if I'm missing something, I glanced over this. What is invoking `run_docker.sh`? The internal jobs that run invoke `addons_gpu.sh`?

Yes, Instead of invoking `addons_gpu.sh` or `addons_cpu.sh` we would request that you instead run one of the below commands depending on which job was running:
```
bash tools/run_docker.sh -c 'make sanity-check'  # Ubuntu Sanity Check
bash tools/run_docker.sh -s linux -d cpu -p py2 -c 'make unit-test'  # Ubuntu Python2
bash tools/run_docker.sh -s linux -d cpu -p py3 -c 'make unit-test'  # Ubuntu Python3
bash tools/run_docker.sh -s linux -d gpu -p py2 -c 'make gpu-unit-test'  # Ubuntu GPU Python2
bash tools/run_docker.sh -s linux -d gpu -p py3 -c 'make gpu-unit-test'  # Ubuntu GPU Python3
```

Is that feasible?

**Comment by facaiy** [comment]
@av8ramit Yeah, as @gunan suggested, 
> i do recommend running docker run from addons_gpu. As long as it is
properly reviewed, i think it should be fine.
> https://github.com/tensorflow/addons/pull/294#issuecomment-506561264

So I think it might be clear to separate CI scripts into two categories:
1. Environment:
    - `run_docker.sh`: prepare the running environment by using docker with appropriate image.
2. Task:
    - `ci_testing/*`:  sanity check, unit test, etc
    - `ci_build/*`: build, package and deploy wheel files.

What do you think? feasible?

**Comment by av8ramit** [comment]
Ah okay, @karmel is it okay if I upgrade the internal scripts to run the following? Just to clarify I'm not an owner of the addons repo so I'd prefer to wait for Karmel to sign off. 

@facaiy is this enough for now?

```
bash tools/run_docker.sh -c 'make sanity-check'  # Ubuntu Sanity Check
bash tools/run_docker.sh -s linux -d cpu -p py2 -c 'make unit-test'  # Ubuntu Python2
bash tools/run_docker.sh -s linux -d cpu -p py3 -c 'make unit-test'  # Ubuntu Python3
bash tools/run_docker.sh -s linux -d gpu -p py2 -c 'make gpu-unit-test'  # Ubuntu GPU Python2
bash tools/run_docker.sh -s linux -d gpu -p py3 -c 'make gpu-unit-test'  # Ubuntu GPU Python3
```

File: tools/run_docker.sh (hunk 1 above)
Comment (av8ramit) [reply tools/run_docker.sh]: Yes each job is run in a separate GCP instance.

**Comment by karmel** [comment]
SGTM-- thanks, all

**Comment by facaiy** [comment]
> @facaiy is this enough for now?

@av8ramit Yes, I think so, Amit. 

Thank you, Amit, Karmel. So we'll merge the PR after Amit confirmed those commands work well in internal scripts, right?

File: tools/run_docker.sh (hunk 1 above)
Comment (facaiy) [reply tools/run_docker.sh]: Awesome, thank you for clarification.

**Comment by av8ramit** [comment]
I've sent @karmel an internal change for the internal build scripts and removes the virtualenv logic. Once merged, I'll update this PR.

**Comment by av8ramit** [comment]
The internal change has been merged and I've kicked off a new round of tests.

**Comment by av8ramit** [comment]
Submitting a fix for the Sanity issue.

**Comment by seanpmorgan** [comment]
> Submitting a fix for the Sanity issue.

@av8ramit @karmel friendly ping to see if we can get the fix in place. Currently we're stuck in limbo as the new commands are active but this PR is not merged so our CI testing is failing across the board.

**Comment by av8ramit** [comment]
It was fixed Monday, sorry I guess someone just needed to rerun it and an owner needs to merge it.

**Pull request merged**